src/main/java/project/OlafApp.java

```diff
@@ -0,0 +1,174 @@
+                    String expression = Parser.parseParameter(command, " ", 1);
+                    Todo newTodo = new Todo(expression);
+                    taskList.addTask(newTodo);
+
+                    storage.saveData(taskList);
+
+                    ui.showNewTaskAddedSuccess(taskList.getTotalNumberOfTasks(),
+                            newTodo, taskList.getTotalNumberOfTasksUndone());
+                } catch (IndexOutOfBoundsException e) {
+                    ui.showFormatError(PrintText.TODO_FORMAT);
+                }
+            } else if (command.toLowerCase().startsWith("deadline")) {
+                try {
+                    String expression = Parser.parseParameter(command, " ", 1);
+                    String description = Parser.parseParameter(expression, "/by", 0);
+
+                    String dateTime = Parser.parseParameter(expression, "/by", 1);
+                    LocalDateTime deadline = Parser.parseDateTimeParameter(dateTime);
+
+                    Deadline newDeadline = new Deadline(description, deadline);
+                    taskList.addTask(newDeadline);
+
+                    storage.saveData(taskList);
+
+                    ui.showNewTaskAddedSuccess(taskList.getTotalNumberOfTasks(),
+                            newDeadline, taskList.getTotalNumberOfTasksUndone());
+                } catch (Exception e) {
+                    // catches both ParseException and IndexOutOfBounds exception
+                    ui.showFormatError(PrintText.DEADLINE_FORMAT);
+                }
+            } else if (command.toLowerCase().startsWith("event")) {
+                try {
+                    String expression = Parser.parseParameter(command, " ", 1);
+                    String description = Parser.parseParameter(expression, "/at", 0);
+
+                    String dateTime = Parser.parseParameter(expression, "/at", 1);
+                    String start = Parser.parseParameter(dateTime, " to ", 0);
+                    String end = Parser.parseParameter(dateTime, " to ", 1);
+
+                    LocalDateTime startDateTime = Parser.parseDateTimeParameter(start);
+                    LocalDateTime endDateTime = Parser.parseDateTimeParameter(end);
+
+                    Event newEvent = new Event(description, startDateTime, endDateTime);
+                    taskList.addTask(newEvent);
+
+                    storage.saveData(taskList);
+
+                    ui.showNewTaskAddedSuccess(taskList.getTotalNumberOfTasks(),
+                            newEvent, taskList.getTotalNumberOfTasksUndone());
+                } catch (Exception e) {
+                    // catches both ParseException and IndexOutOfBounds exception
+                    ui.showFormatError(PrintText.EVENT_FORMAT);
+                }
+            } else if (command.equalsIgnoreCase("help")) {
+                ui.showFormatResponse(PrintText.HELP);
+            } else {
+                ui.showFormatResponse("  Hmm sorry I don't understand :(\n"
```

Comment:
Perhaps this response could be added to your PrintText class?

Reply:
Yup, I agree!